### ISSUE TITLE
Inject[] does not let the interface show up.

### ISSUE THREAD

**Issue by mozesa** [alecthomas/injector#158]
Hello **@alecthomas**,

first of all let me to say _Thank you!_ for this outstanding package.

I try to utilize the feature of `PEP 593` which actually works fine, but _let speak the pictures_.
![image](https://user-images.githubusercontent.com/49973426/95432403-e8a29400-094e-11eb-8cd8-899b5d3ae04c.png)
![image](https://user-images.githubusercontent.com/49973426/95432455-f9eba080-094e-11eb-9e8b-47b1465282dd.png)

As you can see, when I use `Inject[]`, the `PyCharm` doesn't show the interface's methods.

But it works when I use the `Annotated[interface: Type[T], ...]`.
![image](https://user-images.githubusercontent.com/49973426/95433563-661ad400-0950-11eb-88d8-983579fc0b02.png)

Thanks for your help in advance!

**Comment by jstasiak** [comment]
Hey @mozesa. I'll answer in @alecthomas's stead as I maintain this project right now – I think you need to take this up with the PyCharm folks, I don't use it and I'm afraid I can't help much here, everything seems good on our side.

**Comment by mozesa** [comment]
Hello **@jstasiak**,

Thanks for your reply. 
I will create a ticket on PyCharm's side.

I keep open this issue and later will update.

**Comment by jstasiak** [comment]
Sure thing!

**Comment by mozesa** [comment]
[issue](https://youtrack.jetbrains.com/issue/PY-44905) created.

**Comment by JosXa** [comment]
Yes, the JetBrains folks unfortunately lag very much behind updating their type checker. I could [go on another rant](https://github.com/JosXa/tgintegration/issues/15#issue-724228141), but I'll be civil and leave it at that.

The problem is that the `Annotated` feature has not been created with "regular" `__class_getitem__` classmethods, but instead another one of these `_SpecialForm`s. I just spent 30 minutes trying to come up with a workaround, but I couldn't come up with anything better than `def PyCharmCompatibleInject(typ: T) -> T: ...` which could somehow patch that missing behavior, but it looks ugly as hell: `def foo(bar: PChInject(Bar))`. meh.

**Comment by halbow** [comment]
Hi,


I'm still having this issue with Pycharm (mypy works fine) with the Annotation Alias `Inject[Something]`.
A workaround for me was to use `Annotated[Something, _inject_marker]` (using the `Annotated` keyword directly instead of the annotation alias`Inject`).

@jstasiak as it seems Pycharm will not fix this issue soon, would you consider making `_inject_marker` public ? That would allow to have a proper workaround for tools not supporting the alias from [PEP-0593](https://peps.python.org/pep-0593/#aliases-concerns-over-verbosity) yet

**Comment by davidparsson** [comment]
> would you consider making `_inject_marker` public ?

I think that is a reasonable request, allowing a workaround for those that think it's worth it. We should make the `_noinject_marker` public at the same time, and ideally update the documentation.

**Comment by davidparsson** [comment]
I have opened a PR for this in https://github.com/python-injector/injector/pull/269.

**Comment by davidparsson** [comment]
It seems the issue with PyCharm is now fixed.
